src_tests/projectFolderPathUpdater.py

```diff
@@ -0,0 +1,7 @@
+from inspect import getsourcefile
+import os.path as path, sys
+
+current_dir = path.dirname(path.abspath(getsourcefile(lambda: 0)))
+sys.path.insert(0, current_dir[:current_dir.rfind(path.sep)])
```

Comment:
This is kinda scary. I'd say that you should either work with modules (that is not calling `python something.py` but call `python -m folder.file` (without `.py` at the end)) or e.g. configure "source directories" in PyCharm, or whatever.

I did this kind of things more than a few times and I can say honestly – if you can figure out how to do it properly, please do. The current state of things is more likely to bite you in the least expected moment.

Reply:
I created this file in order to be able to run the test scripts outside of PyCharm (in Anaconda console) to reduce CPU usage. In PyCharm I have that pretty much configured and this file would be unneeded, but anaconda doesn't load no project file, thus it has problems with scripts path. How do you propose I proceed in that situation?

Reply:
I'd propose making the project a "proper Python project". What is the actual problem without this file? Is it for running tests? Isn't `python -m unittest` enough?

The "normal" python structure is to have some e.g. `main.py` (`app.py`, whatever) on the top level, then have e.g. `src` directory (module) and import from it and have another `tests` directory where the tests reside. All calls are then done from the top level of the repo (the directory where `main.py` lives in) and thus have all the relative paths set up correctly. Thus, `sys.path` trickery is required only in some dire situations (e.g. setting up a very specific environment). I might be wrong, yet this doesn't seem to be the case.

Reply:
I don't remember how I run tests then, but using `python -m unittest` was fine and didn't require the file we're discussing. I deleted it.